compiler.rb

```diff
@@ -167,7 +169,7 @@
     product_api, provider_config, = \
       Provider::Config.parse(provider_override_path, product_api, version, override_dir)
   end
-  products_for_version.push(product_api)
+  products_for_version.push(definitions: product_api, overrides: provider_config)
```

Comment:
Do we need overrides for anything other than legacy name? Could compute the product/legacy name in compiler and then make it a hash of `product_name -> provider_config`

Or not, because then we would have to sort a hash, so 🤷‍♂

Reply:
While we're only using `legacy_name` now, I'd like to supply the whole override anyways. Conceptually, what I'd like to be available in compiled file is the entire set of MM definitions and not just the API definitions or API definitions+a limited set of overrides.